frontend/ui/src/app/licenses/edit-license.component.ts

```diff
@@ -0,0 +1,267 @@
+import {ApplicationsService} from '../services/applications.service';
+import {Application, ApplicationVersion} from '@glasskube/distr-sdk';
+import {UsersService} from '../services/users.service';
+import dayjs from 'dayjs';
+import {CdkConnectedOverlay, CdkOverlayOrigin} from '@angular/cdk/overlay';
+import {dropdownAnimation} from '../animations/dropdown';
+import {FaIconComponent} from '@fortawesome/angular-fontawesome';
+
+@Component({
+  selector: 'app-edit-license',
+  templateUrl: './edit-license.component.html',
+  imports: [AsyncPipe, AutotrimDirective, ReactiveFormsModule, CdkOverlayOrigin, CdkConnectedOverlay, FaIconComponent],
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => EditLicenseComponent),
+      multi: true,
+    },
+  ],
+  animations: [dropdownAnimation],
+})
+export class EditLicenseComponent implements OnInit, OnDestroy, AfterViewInit, ControlValueAccessor {
+  private injector = inject(Injector);
+  private readonly destroyed$ = new Subject<void>();
+  private readonly applicationsService = inject(ApplicationsService);
+  private readonly usersService = inject(UsersService);
+  applications$ = this.applicationsService.list();
+  customers$ = this.usersService.getUsers().pipe(
+    map((accounts) => accounts.filter((a) => a.userRole === 'customer')),
+    first()
+  );
+
+  private fb = inject(FormBuilder);
+  editForm = new FormGroup({
+    id: new FormControl<string | undefined>(undefined, {nonNullable: true}),
+    name: new FormControl<string | undefined>(undefined, {nonNullable: true, validators: Validators.required}),
+    expiresAt: new FormControl('', {nonNullable: true}),
+    applicationId: new FormControl<string | undefined>(undefined, {nonNullable: true, validators: Validators.required}),
+    includeAllVersions: new FormControl<boolean>(true, {
+      nonNullable: true,
+      validators: Validators.required,
+    }),
+    versions: this.fb.array<boolean>([]),
+    ownerUserAccountId: new FormControl<string | undefined>(undefined, {nonNullable: true}),
+    registry: new FormGroup(
+      {
+        url: new FormControl('', {nonNullable: true}),
+        username: new FormControl('', {nonNullable: true}),
+        password: new FormControl('', {nonNullable: true}),
+      },
+      {
+        validators: (control) => {
+          if (!control.get('url')?.value && !control.get('username')?.value && !control.get('password')?.value) {
+            return null;
+          }
+          if (control.get('url')?.value && control.get('username')?.value && control.get('password')?.value) {
+            return null;
+          }
+          return {
+            required: true,
+          };
+        },
+      }
+    ),
+  });
+  editFormLoading = false;
+  license: WritableSignal<ApplicationLicense | undefined> = signal(undefined);
+  selectedApplication: WritableSignal<Application | undefined> = signal(undefined);
```

Comment:
slightly more concise
```suggestion
  readonly license = signal<ApplicationLicense | undefined>(undefined);
  readonly selectedApplication = signal<Application | undefined>(undefined);
```